internal/provider/workspace_data_source.go

```diff
@@ -0,0 +1,140 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/meshcloud/terraform-provider-meshstack/client"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var (
+	_ datasource.DataSource              = &workspaceDataSource{}
+	_ datasource.DataSourceWithConfigure = &workspaceDataSource{}
+)
+
+func NewWorkspaceDataSource() datasource.DataSource {
+	return &workspaceDataSource{}
+}
+
+type workspaceDataSource struct {
+	client *client.MeshStackProviderClient
+}
+
+func (d *workspaceDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_workspace"
+}
+
+// Schema defines the schema for the data source.
+func (d *workspaceDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Read a single workspace by name.",
+
+		Attributes: map[string]schema.Attribute{
+			"api_version": schema.StringAttribute{
+				MarkdownDescription: "Workspace API version.",
+				Computed:            true,
+			},
+			"kind": schema.StringAttribute{
+				MarkdownDescription: "meshObject type, always `meshWorkspace`.",
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{"meshWorkspace"}...),
+				},
+			},
+
+			"metadata": schema.SingleNestedAttribute{
+				Required: true,
+				Attributes: map[string]schema.Attribute{
+					"name": schema.StringAttribute{
+						MarkdownDescription: "Name of the workspace. Is used to identify the workspace.",
```

Comment:
d: I know we want the property to be named `name` because we draw inspiration from k8s, but the description reads as if the `name` corresponds to what is shown as the workspace name in the panel (and not the identifier).

If we use this description in other places too and want to be consistent now, I'm fine it, just want to share my perspective, because for someone who's familiar with the panel and not so familiar with k8s, this entire name/identifier business is rather confusing.

Reply:
FYI this is how we document it in the API docs 
<img width="1121" height="531" alt="image" src="https://github.com/user-attachments/assets/ff6b2915-bec0-4133-9158-a2727367415b" />